bytecomp/emitcode.ml

```diff
@@ -154,10 +154,21 @@ let slot_for_literal sc =
   enter (Reloc_literal (Symtable.transl_const sc));
   out_int 0
 and slot_for_getglobal id =
-  enter (Reloc_getglobal id);
+  let name = Ident.name id in
+  let reloc_info =
+    if Ident.is_predef id then (Reloc_getpredef (Predef_exn name))
+    else if Ident.global id then (Reloc_getcompunit (Compunit name))
+    else assert false
```

Comment:
Those two `assert false`s (this one and the one in `slot_for_setglobal`)  are a sign of flaw in the propagation of information: the argument of the function should already be a global name, which means changing `Lambda.Psetglobal` and `Lambda.Pgetglobal` to take respectively a `compunit` and `Symtable.Global.t` arguments rather than `Ident.t`s.